cmd/dp-dataset-exporter/main.go

```diff
@@ -2,232 +2,304 @@ package main
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"os/signal"
 	"syscall"
 	"time"
 
-	"github.com/ONSdigital/go-ns/clients/dataset"
-	filterHealthCheck "github.com/ONSdigital/go-ns/clients/filter"
-	"github.com/ONSdigital/go-ns/healthcheck"
-	"github.com/ONSdigital/go-ns/log"
-	"github.com/ONSdigital/go-ns/rchttp"
+	"github.com/ONSdigital/dp-api-clients-go/dataset"
+	filterCli "github.com/ONSdigital/dp-api-clients-go/filter"
+	"github.com/ONSdigital/dp-api-clients-go/health"
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	kafka "github.com/ONSdigital/dp-kafka"
+	vault "github.com/ONSdigital/dp-vault"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
 
 	"github.com/ONSdigital/dp-dataset-exporter/config"
 	"github.com/ONSdigital/dp-dataset-exporter/errors"
 	"github.com/ONSdigital/dp-dataset-exporter/event"
+	"github.com/ONSdigital/dp-dataset-exporter/file"
 	"github.com/ONSdigital/dp-dataset-exporter/filter"
 	"github.com/ONSdigital/dp-dataset-exporter/initialise"
 	"github.com/ONSdigital/dp-dataset-exporter/schema"
+	"github.com/ONSdigital/go-ns/server"
+)
+
+var (
+	// BuildTime represents the time in which the service was built
+	BuildTime string
+	// GitCommit represents the commit (SHA-1) hash of the service that is running
+	GitCommit string
+	// Version represents the version of the service that is running
+	Version string
 )
 
 func main() {
+	ctx := context.Background()
 	log.Namespace = "dp-dataset-exporter"
-	log.Info("Starting dataset exporter", nil)
+	log.Event(ctx, "Starting dataset exporter")
 
 	signals := make(chan os.Signal, 1)
 	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
 
 	cfg, err := config.Get()
-	exitIfError(err)
+	exitIfError(ctx, err)
 
-	log.Info("loaded config", log.Data{"config": cfg})
+	log.Event(ctx, "loaded config", log.Data{"config": cfg})
 
 	// a channel used to signal when an exit is required
 	errorChannel := make(chan error)
 
+	// serviceList keeps track of what dependency services have been initialised
 	var serviceList initialise.ExternalServiceList
 
-	kafkaBrokers := cfg.KafkaAddr
-	kafkaConsumer, err := serviceList.GetConsumer(kafkaBrokers, cfg)
-	logIfError(err)
+	// Create kafka Consumer
+	kafkaConsumer, err := serviceList.GetConsumer(cfg)
+	exitIfError(ctx, err)
```

Comment:
I don't think we should  be forcing a shutdown if we are unable to connect to kafka?

Reply:
At creation time, Kafka will only return an error if the channel validation fails (unrecoverable error). If the broker cannot be contacted, it will not error and the healthcheck will initialise the producer/consumer once the broker(s) is/are available.